### PR TITLE
Jormun: Add cycle lane length

### PR DESCRIPTION
**Pull request by AurelienLP**
We want to know the percentage of cycle path used for a bike journey.
For now, only asgard is able to retrieve these information. The aim of this PR is to compute and expose these information in the section part of the navitia response

### CODE REVIEW

File: source/jormungandr/tests/direct_path_asgard_integration_tests.py [hunk 1]
```diff
@@ -140,6 +160,7 @@ def test_journey_with_bike_direct_path(self):
         assert response['journeys'][0]['durations']['car'] == 500
         assert response['journeys'][0]['durations']['total'] == 500
         assert response['journeys'][0]['distances']['car'] == 50
+        assert response['journeys'][0]['sections'][0]['cycle_path_percentage'] == 0.0
```
Comment (benoit-bst) [inline source/jormungandr/tests/direct_path_asgard_integration_tests.py]: Why it is present inside a car section ? The field is not optional ?

**Review by benoit-bst — APPROVED** [review]

File: source/jormungandr/tests/direct_path_asgard_integration_tests.py (hunk 1 above)
Comment (AurelienLP) [reply source/jormungandr/tests/direct_path_asgard_integration_tests.py]: Fixed that! Thank you :)

File: source/jormungandr/tests/direct_path_asgard_integration_tests.py [hunk 2]
```diff
@@ -149,6 +170,7 @@ def test_journey_with_bike_direct_path(self):
         assert response['journeys'][1]['durations']['total'] == 1000
         assert response['journeys'][1]['distances']['bike'] == 100
         assert response['journeys'][1]['duration'] == 1000
+        assert response['journeys'][1]['sections'][0]['cycle_lane_length'] == 30.0
```
Comment (mbounabi) [inline source/jormungandr/tests/direct_path_asgard_integration_tests.py]: It is cast as 'int' in 'add_cycle_lane_length', isn't it?

File: source/jormungandr/jormungandr/interfaces/v1/serializer/journey.py [hunk 3]
```diff
@@ -312,6 +312,15 @@ def get_ridesharing_journeys(self, obj):
             return None
         return JourneySerializer(obj.ridesharing_journeys, display_none=False, many=True).data
 
+    cycle_lane_length = jsonschema.MethodField(
+        schema_type=int, display_none=False, description='Length of the section on cycle lanes'
+    )
+
+    def get_cycle_lane_length(self, obj):
+        if not hasattr(obj, 'cycle_lane_length') or not obj.cycle_lane_length:
```
Comment (pbougue) [inline source/jormungandr/jormungandr/interfaces/v1/serializer/journey.py]: minor: Don't you want to output `0` when you have the information and you know it's 0 ?
With this it's not possible to differentiate `0` vs. `no information available/accurate`.
Don't know if it's easily doable to differentiate though, depending on the outputs of connectors (and what protobuf allows).

File: source/jormungandr/tests/direct_path_asgard_integration_tests.py (hunk 2 above)
Comment (pbougue) [reply source/jormungandr/tests/direct_path_asgard_integration_tests.py]: minor (for readability): So shouldn't you have :
```suggestion
        assert response['journeys'][1]['sections'][0]['cycle_lane_length'] == 30
```

**Review by pbougue — APPROVED** [review]
:+1: I guess geovelo will be the next one

File: source/jormungandr/jormungandr/interfaces/v1/serializer/journey.py (hunk 3 above)
Comment (AurelienLP) [reply source/jormungandr/jormungandr/interfaces/v1/serializer/journey.py]: That's the behavior I wanted. Did it with `PbIntField(display_none=False)`

**Review by benoit-bst — APPROVED** [review]

**Review by xlqian — APPROVED** [review]

**Review by mbounabi — APPROVED** [review]

**Pull request merged**